ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Ledger.hs

```diff
@@ -72,13 +73,25 @@ instance CardanoHardForkConstraints c
 
   decodeCanonicalTxIn = CardanoTxIn <$> Core.fromEraCBOR @(ShelleyEra c)
 
+-- Unpacking the fields of the era-specific TxOuts could save a chunk of memory.
+-- However, unpacking of sum types is only possible on @ghc-9.6.1@ and later, so
+-- before @ghc-9.6.1@ we only unpack the TxOuts for ears before Alonzo.
 data CardanoTxOut c =
-    ShelleyTxOut !(Value (LedgerState (ShelleyBlock (TPraos c) (ShelleyEra c))))
-  | AllegraTxOut !(Value (LedgerState (ShelleyBlock (TPraos c) (AllegraEra c))))
-  | MaryTxOut    !(Value (LedgerState (ShelleyBlock (TPraos c) (MaryEra c))))
+#if MIN_VERSION_GLASGOW_HASKELL(9,6,1,0)
```

Comment:
We usually use `__GLASGOW_HASKELL__ > 906`, couldn't we use it also here or is it due to the patch version? I don't think `__GLASGOW_HASKELL__` has a minor field

Reply:
The `z` in `MIN_VERSION_GLASGOW_HASKELL(x,y,z,z')` is important here. It's only from `ghc-9.6.1` onwards that we have this feature. `__GLASGOW_HASKELL__ > xyy` wouldn't let us put a bound on the `z`

Sources:
https://downloads.haskell.org/ghc/latest/docs/users_guide/phases.html?highlight=cpp#standard-cpp-macros
https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/pragmas.html#unpack-pragma

Reply:
Alright, clear now, thanks!